src/sensor/pingsensor.cpp

```diff
@@ -14,6 +14,21 @@ PingSensor::PingSensor()
     connect(dynamic_cast<PingParserExt*>(_parser), &PingParserExt::parseError, this, &PingSensor::parserErrorsUpdate);
 }
 
+void PingSensor::connectLink(const LinkConfiguration conConf, const LinkConfiguration& logConf)
+{
+    Sensor::connectLink(conConf, logConf);
+    if (_parser) {
+        connect(link(), &AbstractLink::newData, this, [this](QByteArray data) {
```

Comment:
why are you not using the newMessage signal and doing the parser job here ? This is not explicit described in the commit body or in comments. How this helps ?
Also, should this be fixed in the `parseBuffer` function ?

Reply:
> removed use of newMessage signal from parser, and parse the buffer/handle the message synchronously

I'm sorry it's not clear, I'm trying to handle the message immediately, rather than waiting for the newMessage signal to propagate. (I'm a noob at threading and signal/slots, my understanding is not solid).